src/plugins/orangehrmDashboardPlugin/Service/EmployeeTimeAtWorkService.php

```diff
@@ -0,0 +1,287 @@
+                $attendanceRecord->getDecorator()->getPunchOutUTCDate(),
+                $attendanceRecord->getDecorator()->getPunchOutUTCTime(),
+                $attendanceRecord->getDecorator()->getPunchOutUserDate(),
+                $attendanceRecord->getDecorator()->getPunchOutUserTime(),
+                $attendanceRecord->getPunchOutTimeOffset()
+            );
+        }
+        return [
+            'state' => $actionDetails->getState(),
+            'utcDate' => $actionDetails->getUtcDate(),
+            'utcTime' => $actionDetails->getUtcTime(),
+            'userDate' => $actionDetails->getUserDate(),
+            'userTime' => $actionDetails->getUserTime(),
+            'timezoneOffset' => $actionDetails->getTimezoneOffset()
+        ];
+    }
+
+    /**
+     * @param int $empNumber
+     * @param DateTime $startUTCDateTime
+     * @return int total time will be returned in minutes
+     * @throws Exception
+     */
+    private function getTotalTimeForGivenDate(int $empNumber, DateTime $startUTCDateTime): int
+    {
+        $totalTime = 0;
+        $endUTCDateTime = (clone $startUTCDateTime)->add(new DateInterval('P1D'));
+        $attendanceRecords = $this->getEmployeeTimeAtWorkDao()
+            ->getAttendanceRecordsByEmployeeAndDate(
+                $empNumber,
+                $startUTCDateTime,
+                $endUTCDateTime
+            );
+        /**
+         * No attendance records found for given day, check for empty array []
+         */
+        if (!$attendanceRecords) {
+            return $totalTime;
+        }
+        foreach ($attendanceRecords as $attendanceRecord) {
+            if ($attendanceRecord->getState() === self::STATE_PUNCHED_OUT) {
+                $punchInUtcDateTime = $this->getDateTimeInUTC($attendanceRecord->getPunchInUtcTime());
+                $punchOutUtcDateTime = $this->getDateTimeInUTC($attendanceRecord->getPunchOutUtcTime());
+                if ($punchInUtcDateTime < $startUTCDateTime && $punchOutUtcDateTime > $endUTCDateTime) {
+                    $totalTime += $this->getTimeDifference($endUTCDateTime, $startUTCDateTime);
+                } elseif ($punchInUtcDateTime < $startUTCDateTime && $punchOutUtcDateTime > $startUTCDateTime) {
+                    $totalTime += $this->getTimeDifference($startUTCDateTime, $punchOutUtcDateTime);
+                } elseif ($punchInUtcDateTime < $endUTCDateTime && $punchOutUtcDateTime > $endUTCDateTime) {
+                    $totalTime += $this->getTimeDifference($punchInUtcDateTime, $endUTCDateTime);
+                } else {
+                    $totalTime += $this->getTimeDifference($punchOutUtcDateTime, $punchInUtcDateTime);
+                }
+            }
+        }
+        return $totalTime;
+    }
+
+    /**
+     * @param DateTime $date
+     * @return array  eg:- array(if monday as first day in config => '2021-12-13', '2021-12-19')
+     */
+    private function getWeekBoundaryForGivenDate(DateTime $date): array
```

Comment:
will reuse the attendance module method if possible. if this method not same as attendance, no need changes